### ISSUE TITLE
[BUG] Custom Skull Overrides using block data instead of item data

### ISSUE THREAD

**Issue by PhanaticD** [LoneDev6/ItemsAdder#283]
example items.yml in item packs:
```
info:
  namespace: slimefun
items:
  carbonado:
    resource:
      material: PLAYER_HEAD
      generate: true
      model_id: 4000
      textures:
      - slimefun/carbonado.png
```

These are the two files that are created by IA
\plugins\ItemsAdder\data\resource_pack\assets\minecraft\models\itemplayer_head.json
```
{
  "parent": "block/player_head",
  "overrides": [
    {
      "predicate": {
        "custom_model_data": 4000
      },
      "model": "slimefun:auto_generated/carbonado"
    }
  ]
}
```
first problem is this should be "item/template_skull" not "block/player_head"
second problem happens if you set generate: false in the item pack, it still sets the path to auto_generated/carbonado instead of items/carbonado

AND

\plugins\ItemsAdder\data\resource_pack\assets\slimefun\models\auto_generated\carbonado.json
```
{
  "parent": "item/generated",
  "textures": {
    "all": "slimefun:slimefun/carbonado",
    "particle": "slimefun:slimefun/carbonado"
  }
}
```
the problem with this file is I assume its still think its using the block texture, but instead it should be just "layer0": "slimefun:slimefun/carbonado"

I have spent a long time trying to figure this issue out and tested that these changes did fix the custom texture to show up, please fix :D

**Comment by LoneDev6** [comment]
Hello!
You can do this:
```Yaml
info:
  namespace: slimefun
items:
  carbonado:
    resource:
      material: PLAYER_HEAD
      generate: false
      model_id: 4000
      model_path: item/carbonado
```


and create the carbonado.json file inside assets\slimefun\models\item folder of your pack:
```Json
{
  "parent": "item/template_skull",
  "textures": {
    "layer0": "slimefun:item/carbonado",
  }
}
```

(texture file would be inside assets\slimefun\textures\item)

**Comment by PhanaticD** [comment]
shouldnt it still be "item/template_skull" not "block/player_head" anyways for that file?

**Comment by LoneDev6** [comment]
> 
> 
> isnt that still just overriding all the mistakes? instead of actually making it properly (in case you do use generate: true)

Yep I'm going to fix that :D

Now material: PLAYER_HAD would automatically generate the correct parent.


```YAML
info:
  namespace: slimefun
items:
  carbonado:
    resource:
      material: PLAYER_HEAD
      generate: true
      model_id: 4000
      textures:
      - slimefun/carbonado.png
```

**Comment by LoneDev6** [comment]
comment on this if after the update it's not fixed, i will reopen

**Comment by PhanaticD** [comment]
this still is happening with glass panes and wither skulls, and probably other blocks that are also items

**Comment by LoneDev6** [comment]
> 
> 
> this still is happening with glass panes and wither skulls, and probably other blocks that are also items

Thanks for the info!
Are you reporting this because you need to have this working also on glass panes and wither skulls or you reported it just because you noticed?

thanks again

**Comment by PhanaticD** [comment]
I do need it (to add the slimefun texture pack) wither skulls, glass panes, comparators, powered and activator rails are all items used by slimefun that have custom model ID, but when created using itemsadder the wrong parent is applied, it does the block texture parent instead of the held item parent

**Comment by LoneDev6** [comment]
https://www.spigotmc.org/resources/addon-slimefun-compatibility-for-itemsadder.83877/

**Comment by LoneDev6** [comment]
I fixed the parenting issue completely in 2.0.46, let me know.
Now it will apply the correct parent, textures and display attributes to the generated vanilla json file
